ext/http/response_body.rs

```diff
@@ -62,19 +101,41 @@ impl Future for CompletionHandle {
   }
 }
 
+trait PollFrame: Unpin {
+  fn poll_frame(
+    self: Pin<&mut Self>,
+    cx: &mut std::task::Context<'_>,
+  ) -> std::task::Poll<ResponseStreamResult>;
+
+  fn size_hint(&self) -> SizeHint;
+}
+
+#[derive(PartialEq, Eq)]
+pub enum Compression {
+  None,
+  GZip,
+}
+
+pub enum ResponseStream {
+  /// A resource stream, piped in fast mode.
```

Comment:
Add a comment that this kind doesn't involve going to JS to obtain more data and always stays in Rust while streaming